onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -253,7 +253,39 @@ void InputVariadicSelector::UpdateBuilder(NodesToOptimizeIndicesBuilder& builder
   builder.num_input_defs = 1;  // set to 1 as the first input is variadic
 }
 
-void OutputVariadicSelector::UpdateBuilder(NodesToOptimizeIndicesBuilder& builder) const {
+bool SplitNodeGroupSelector::Check(const GraphViewer& graph_viewer,
+                                   const Node& node,
+                                   const std::vector<const Node*>& dq_nodes,
+                                   const std::vector<const Node*>& q_nodes) const {
+  if (!CheckQDQNodes(graph_viewer, node, dq_nodes, q_nodes, 1)) {
+    return false;
+  }
+
+  auto get_const_initializer = [&graph_viewer](const std::string& initializer_name) {
+    return graph_viewer.GetConstantInitializer(initializer_name, true);
+  };
+
+  const Node& dq_node = *dq_nodes.front();
+  int32_t dt_input = dq_node.InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+
+  // All Q outputs should have same data type and (optionally) equal quantization parameters as the input.
+  for (size_t q_idx = 0; q_idx < q_nodes.size(); q_idx++) {
+    const Node& q_node = *q_nodes[q_idx];
+
+    if (dt_input != q_node.OutputDefs()[0]->TypeAsProto()->tensor_type().elem_type()) {
+      return false;
+    }
+
+    if (req_equal_quant_params_ &&
```

Comment:
When would we not require the values to match?

Reply:
QNN EP does not require the values to match. The reason is because the quantization script can create valid QDQ models in which the values do not match. I tried to provide an example of such a model in this comment: https://github.com/microsoft/onnxruntime/pull/18394#discussion_r1389910846

To summarize that example, a Clip after a Split output will be "merged" into the output Q node.

This float32 graph
```
X -> Split -> Clip -> S0
          \-> S1
```

Is converted to this QDQ graph. where `Q0*` has different quantization params due to Clip.
```
X -> DQin -> Split -> Q0* -> DQa -> S0
                  \-> Q1 -> DQb -> S1
```